pxr/imaging/hdMtlx/hdMtlx.cpp

```diff
@@ -277,7 +277,7 @@ _GatherUpstreamNodes(
 
     // Initilize the mxNodeGraph if needed
     if (!(*mxNodeGraph)) {
-        const std::string & nodeGraphName  = hdNodePath.GetParentPath().GetName();
+        std::string nodeGraphName = hdNodePath.GetParentPath().GetName() + "_Graph";
```

Comment:
As a suggestion here, there is a `makeUniqueChildName()` method which can be called on `mxDoc`, and should be more robust than appendig "_Graph" -- in case this is also a duplicate.

Reply:
Seems like you mean `createValidChildName`. Yeah we can use it here